File: compat/src/market/kotlin/ohos/app/ContextDeal.java
package ohos.app;

import java.io.File;

import ohos.aafwk.ability.IAbilityConnection;
import ohos.aafwk.ability.IAbilityStartCallback;
import ohos.aafwk.ability.startsetting.AbilityStartSetting;
import ohos.aafwk.content.Intent;
import ohos.app.dispatcher.TaskDispatcher;
import ohos.app.dispatcher.task.TaskPriority;
import ohos.bundle.AbilityInfo;
import ohos.bundle.ApplicationInfo;
import ohos.bundle.ElementName;
import ohos.bundle.HapModuleInfo;
import ohos.bundle.IBundleManager;
import ohos.global.configuration.Configuration;
import ohos.global.resource.ResourceManager;
import ohos.global.resource.solidxml.Pattern;
import ohos.global.resource.solidxml.Theme;
import ohos.utils.net.Uri;

public class ContextDeal implements Context {

    public ContextDeal(android.content.Context context, ClassLoader classLoader) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public TaskDispatcher createParallelTaskDispatcher(String s, TaskPriority taskPriority) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public TaskDispatcher createSerialTaskDispatcher(String s, TaskPriority taskPriority) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public TaskDispatcher getGlobalTaskDispatcher(TaskPriority taskPriority) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public TaskDispatcher getMainTaskDispatcher() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public TaskDispatcher getUITaskDispatcher() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ApplicationInfo getApplicationInfo() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ProcessInfo getProcessInfo() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public AbilityInfo getAbilityInfo() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ResourceManager getResourceManager() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getPreferencesDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getDatabaseDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getDistributedDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void switchToDeviceEncryptedStorageContext() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void switchToCredentialEncryptedStorageContext() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean isDeviceEncryptedStorage() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean isCredentialEncryptedStorage() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int verifyCallingPermission(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int verifySelfPermission(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int verifyCallingOrSelfPermission(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int verifyPermission(String s, int i, int i1) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ClassLoader getClassloader() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Context getApplicationContext() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void terminateAbility() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void terminateAbility(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void displayUnlockMissionMessage() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void lockMission() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void unlockMission() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getLocalClassName() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ElementName getElementName() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ElementName getCallingAbility() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getCallingBundle() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean stopAbility(Intent intent) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void startAbility(Intent intent, int i) {
        throw new RuntimeException("Stub!");
    }

    //<editor-fold desc="api level 7">
    @Override
    public void startAbility(Intent intent, AbilityStartSetting abilityStartSetting, IAbilityStartCallback iAbilityStartCallback) {
        throw new RuntimeException("Stub!");
    }
    //</editor-fold>

    @Override
    public void startAbility(Intent intent, int i, AbilityStartSetting abilityStartSetting) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void startAbilities(Intent[] intents) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Context createBundleContext(String s, int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean canRequestPermission(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void requestPermissionsFromUser(String[] strings, int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean connectAbility(Intent intent, IAbilityConnection iAbilityConnection) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void disconnectAbility(IAbilityConnection iAbilityConnection) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setDisplayOrientation(AbilityInfo.DisplayOrientation displayOrientation) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public IBundleManager getBundleManager() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getBundleName() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getBundleCodePath() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getBundleResourcePath() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getDataDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getCacheDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getCodeCacheDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File[] getExternalMediaDirs() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getNoBackupFilesDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getFilesDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getDir(String s, int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getExternalCacheDir() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File[] getExternalCacheDirs() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File getExternalFilesDir(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public File[] getExternalFilesDirs(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean deleteFile(String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public IAbilityManager getAbilityManager() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean terminateAbilityResult(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int getDisplayOrientation() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setShowOnLockScreen(boolean b) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setWakeUpScreen(boolean b) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void restart() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setTransitionAnimation(int i, int i1) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public boolean isUpdatingConfigurations() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setTheme(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Theme getTheme() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int getThemeId() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setPattern(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Pattern getPattern() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getAppType() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public ResourceManager getResourceManager(Configuration configuration) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Object getLastStoredDataWhenConfigChanged() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void printDrawnCompleted() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyPermission(String s, String s1) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyUriPermission(Uri uri, int i, String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyCallerPermission(String s, String s1) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyCallerUriPermission(Uri uri, int i, String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyPermission(String s, int i, int i1, String s1) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyUriPermission(Uri uri, int i, int i1, int i2, String s) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void compelVerifyUriPermission(Uri uri, String s, String s1, int i, int i1, int i2, String s2) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int getColor(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getString(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getString(int i, Object... objects) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String[] getStringArray(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int[] getIntArray(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public String getProcessName() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Context getAbilityPackageContext() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public HapModuleInfo getHapModuleInfo() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public Uri getCaller() {
        throw new RuntimeException("Stub!");
    }

    @Override
    public void setColorMode(int i) {
        throw new RuntimeException("Stub!");
    }

    @Override
    public int getColorMode() {
        throw new RuntimeException("Stub!");
    }

    public void setApplication(Application application) {
        throw new RuntimeException("Stub!");
    }
}

File: app/src/market/kotlin/ohos/app/ContextDeal.kt
package ohos.app

import ohos.aafwk.ability.IAbilityConnection
import ohos.aafwk.ability.IAbilityStartCallback
import ohos.aafwk.ability.startsetting.AbilityStartSetting
import ohos.aafwk.content.Intent
import ohos.app.dispatcher.TaskDispatcher
import ohos.app.dispatcher.task.TaskPriority
import ohos.bundle.AbilityInfo
import ohos.bundle.ApplicationInfo
import ohos.bundle.ElementName
import ohos.bundle.HapModuleInfo
import ohos.bundle.IBundleManager
import ohos.global.configuration.Configuration
import ohos.global.resource.ResourceManager
import ohos.global.resource.solidxml.Pattern
import ohos.global.resource.solidxml.Theme
import ohos.utils.net.Uri
import java.io.File

class ContextDeal(context: android.content.Context, classLoader: ClassLoader) : Context {

    init {
        throw RuntimeException("Stub!")
    }

    override fun createParallelTaskDispatcher(s: String, taskPriority: TaskPriority): TaskDispatcher {
        throw RuntimeException("Stub!")
    }

    override fun createSerialTaskDispatcher(s: String, taskPriority: TaskPriority): TaskDispatcher {
        throw RuntimeException("Stub!")
    }

    override fun getGlobalTaskDispatcher(taskPriority: TaskPriority): TaskDispatcher {
        throw RuntimeException("Stub!")
    }

    override fun getMainTaskDispatcher(): TaskDispatcher {
        throw RuntimeException("Stub!")
    }

    override fun getUITaskDispatcher(): TaskDispatcher {
        throw RuntimeException("Stub!")
    }

    override fun getApplicationInfo(): ApplicationInfo {
        throw RuntimeException("Stub!")
    }

    override fun getProcessInfo(): ProcessInfo {
        throw RuntimeException("Stub!")
    }

    override fun getAbilityInfo(): AbilityInfo {
        throw RuntimeException("Stub!")
    }

    override fun getResourceManager(): ResourceManager {
        throw RuntimeException("Stub!")
    }

    override fun getResourceManager(configuration: Configuration): ResourceManager {
        throw RuntimeException("Stub!")
    }

    override fun getBundleName(): String {
        throw RuntimeException("Stub!")
    }

    override fun getBundleCodePath(): String {
        throw RuntimeException("Stub!")
    }

    override fun getBundleResourcePath(): String {
        throw RuntimeException("Stub!")
    }

    override fun getDataDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getCacheDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getCodeCacheDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getExternalMediaDirs(): Array<File> {
        throw RuntimeException("Stub!")
    }

    override fun getNoBackupFilesDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getFilesDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getDir(s: String, i: Int): File {
        throw RuntimeException("Stub!")
    }

    override fun getExternalCacheDir(): File {
        throw RuntimeException("Stub!")
    }

    override fun getExternalCacheDirs(): Array<File> {
        throw RuntimeException("Stub!")
    }

    override fun getExternalFilesDir(s: String): File {
        throw RuntimeException("Stub!")
    }

    override fun getExternalFilesDirs(s: String): Array<File> {
        throw RuntimeException("Stub!")
    }

    override fun deleteFile(s: String): Boolean {
        throw RuntimeException("Stub!")
    }

    override fun getAbilityManager(): IAbilityManager {
        throw RuntimeException("Stub!")
    }

    override fun terminateAbilityResult(i: Int): Boolean {
        throw RuntimeException("Stub!")
    }

    override fun getDisplayOrientation(): Int {
        throw RuntimeException("Stub!")
    }

    override fun setShowOnLockScreen(b: Boolean) {
        throw RuntimeException("Stub!")
    }

    override fun setWakeUpScreen(b: Boolean) {
        throw RuntimeException("Stub!")
    }

    override fun getLastStoredDataWhenConfigChanged(): Any {
        throw RuntimeException("Stub!")
    }

    override fun printDrawnCompleted() {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyPermission(s: String, s1: String) {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyUriPermission(uri: Uri, i: Int, s: String) {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyCallerPermission(s: String, s1: String) {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyCallerUriPermission(uri: Uri, i: Int, s: String) {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyPermission(s: String, i: Int, i1: Int, s1: String) {
        throw RuntimeException("Stub!")
    }

    override fun compelVerifyUriPermission(uri: Uri, i: Int, i1: Int, i2: Int, s: String) {
        throw RuntimeException("Stub!")
    }

    override fun verifySelfPermission(s: String): Int {
        throw RuntimeException("Stub!")
    }

    override fun verifyCallingPermission(s: String): Int {
        throw RuntimeException("Stub!")
    }

    override fun verifyCallingOrSelfPermission(s: String): Int {
        throw RuntimeException("Stub!")
    }

    override fun verifyPermission(s: String, i: Int, i1: Int): Int {
        throw RuntimeException("Stub!")
    }

    override fun getBundleManager(): IBundleManager {
        throw RuntimeException("Stub!")
    }

    override fun requestPermissionsFromUser(array: Array<String>, i: Int, permissionRequestResult: PermissionRequestResult) {
        throw RuntimeException("Stub!")
    }

    override fun canRequestPermission(s: String): Boolean {
        throw RuntimeException("Stub!")
    }

    override fun startAbility(intent: Intent, i: Int) {
        throw RuntimeException("Stub!")
    }

    override fun startAbility(intent: Intent, abilityStartSetting: AbilityStartSetting, i: Int) {
        throw RuntimeException("Stub!")
    }

    override fun startAbilityForResult(intent: Intent, i: Int, iAbilityStartCallback: IAbilityStartCallback) {
        throw RuntimeException("Stub!")
    }

    override fun startAbilityForResult(intent: Intent, abilityStartSetting: AbilityStartSetting, i: Int, iAbilityStartCallback: IAbilityStartCallback) {
        throw RuntimeException("Stub!")
    }

    override fun stopAbility(intent: Intent): Boolean {
        throw RuntimeException("Stub!")
    }

    override fun connectAbility(intent: Intent, iAbilityConnection: IAbilityConnection): Boolean {
        throw RuntimeException("Stub!")
    }

    override fun disconnectAbility(iAbilityConnection: IAbilityConnection) {
        throw RuntimeException("Stub!")
    }

    override fun getPattern(): Pattern {
        throw RuntimeException("Stub!")
    }

    override fun getTheme(): Theme {
        throw RuntimeException("Stub!")
    }

    override fun getTheme(resourceManager: ResourceManager): Theme {
        throw RuntimeException("Stub!")
    }

    override fun setPattern(pattern: Pattern) {
        throw RuntimeException("Stub!")
    }

    override fun setTheme(theme: Theme) {
        throw RuntimeException("Stub!")
    }

    override fun getHapModuleInfo(): HapModuleInfo {
        throw RuntimeException("Stub!")
    }

    override fun getElementName(): ElementName {
        throw RuntimeException("Stub!")
    }

    override fun getColorMode(): Int {
        throw RuntimeException("Stub!")
    }

    fun setApplication(application: Application) {
        throw RuntimeException("Stub!")
    }
}